src/types/select.js

```diff
@@ -797,17 +797,17 @@ export default {
 	},
 
 	deMap: function (obj) {
-		if(!obj){
+		if (!obj) {
 			return null;
 		}
-		return obj.value || null;
+		return obj.value;
 	},
 
 	isEmpty: function () {
 		const self = this,
 			data = self.deMap(self.item);
 
-		return data !== 0 && !data;
+		return data === 0 || data === undefined || data === null || data === '';
```

Comment:
This is changing the logic to say a value of `0` is empty, was that intended?.  Also `!data` already covers all of the other checks, are you doing these explicit falsey checks to allow a specific falsey value to be considered as not empty?